Analytics/resources/themes/get_themes.py

```diff
@@ -0,0 +1,68 @@
+from http import HTTPStatus
+
+from flask_jwt_extended import jwt_required
+from flask_restful import Resource
+from flask_restful import reqparse
+
+from models.theme import Theme
+
+
+class GetThemes(Resource):
+    """
+    Endpoint used to fetch themes from the database table 'theme'
+    """
+
+    def __init__(self) -> None:
+        """
+        Instantiates the endpoint to get a unit from the database table unit.
+        """
+        self.reqparser = reqparse.RequestParser()
+        self.reqparser.add_argument('limit', required=False, store_missing=False, type=int)
+        self.reqparser.add_argument('name', required=False, store_missing=False, type=str)
+        self.reqparser.add_argument('id', required=False, store_missing=False, type=str)
+
+    @jwt_required
+    def get(self) -> ([Theme], HTTPStatus):
+        """
+        Endpoint used to fetch themes from the database
+        Parameters can be passed using a POST request that contains a JSON with the following fields:
+        :param limit: the maximum number of entries to return
+        :param name: themes name
+        :param id: the themes identification number
+        :type limit: int
+        :type name: str
+        :type id: str
+
+        :return: a list of Theme/s and an HTTPStatus code of 200 on success otherwise an JSON with an error message
+                 and appropriate http status
+        """
+        # Get arguments passed in POST request
+        args = self.reqparser.parse_args()
+
+        # pdb.set_trace()
+        # fetch themes using arguments passed in post request
+        themes = []
+        if "id" in args:
+            theme = Theme.get_by_id(args["id"])
+            if theme:
```

Comment:
Consider refactoring this. Don't need two copies of
```python
            if theme:
                themes.append(theme.json())
```
Rather do:

```python
if "name" in args:
    theme = Theme.get_by_name(args["name"])
elif "id" in args:
    theme = Theme.get_by_id(args["id"])

theme.append(theme.json())
```
Or something like that.